probe-rs/src/flashing/erase.rs

```diff
@@ -37,6 +37,14 @@ pub fn erase_all(session: &mut Session) -> Result<(), FlashError> {
         }
     }
 
+    if session.has_custom_erase_all() {
```

Comment:
I am not sure which erase method we want to prefer for erasing, but we should def. not erase twice.
Also, is it necessary to pass permissions here? Wouldn't this be a higher level thing? I see that we pass this to `erase_all` down the stack. But can we get rid of that? It seems unnecessary.

Reply:
We could probably push it down a few levels, but don't we want permissions in the CustomEraseSequence API? 
Also does it make sense to have this `custom_erase_all` method `Session` It feels a bit out of place but at the same time its the only common location that both the eraser and flasher/loader have access too that I could see.

Reply:
Yeah, it's a bit odd I guess, but the only other place that might work I can imagine is the `DebugSequence` enum but that might make for a weird API for actually retrieving the custom erase sequence handle.

Reply:
Yes i agree it would be a bit weird.  I'll keep it like this for now, but remove Permission from `CustomEraseSequence` and instead enforce it on a higher level.